Changing gems in build_config causes problems by leftover references

When I change the included gems in build_config, I need to do a deep_clean. If I don't, the new gem is not included. If I removed a gem, part of the references seem to be still there and the C compiler crashes during 'make test'.

````
...
LD    build/test/bin/mrbtest 
Undefined symbols for architecture x86_64:
  "_GENERATED_TMP_mrb_ruby_extension_example_gem_test", referenced from:
      _mrbgemtest_init in mrbtest.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
Undefined symbols for architecture x86_64:
  "_GENERATED_TMP_mrb_ruby_extension_example_gem_test", referenced from:
      _mrbgemtest_init in mrbtest.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
rake aborted!
Command Failed: [gcc  -o "/Users/hlogmans/development/mruby/usb2/build/test/bin/mrbtest" "/Users/hlogmans/development/mruby/usb2/build/test/mrbgems/mruby-test/driver.o" "/Users/hlogmans/development/mruby/usb2/build/test/mrbgems/mruby-test/mrbtest.o" "/Users/hlogmans/development/mruby/usb2/build/test/mrbgems/mruby-test/mrbtest.a" "/Users/hlogmans/development/mruby/usb2/build/test/lib/libmruby_core.a" "/Users/hlogmans/development/mruby/usb2/build/test/lib/libmruby.a"  -lm -lreadline -lncurses ]

`````

If I run `rake  deep_clean` and then build, it runs fine. This is nowhere documents and by accident I noticed the deep_clean command...

Using lastest mruby from git.

--- a/mrbgems/mruby-test/mrbgem.rake
+++ b/mrbgems/mruby-test/mrbgem.rake
@@ -146,6 +146,19 @@ MRuby::Gem::Specification.new('mruby-test') do |spec|
   end
 
   init = "#{spec.dir}/init_mrbtest.c"
+
+  # store the last gem selection and make the re-build
+  # of the test gem depending on a change to the gem
+  # selection
+  active_gems = "#{build_dir}/active_gems.lst"
+  FileUtils.mkdir_p File.dirname(active_gems)
+  open(active_gems, 'w+') do |f|
+    build.gems.each do |g|
+      f.puts g.name
+    end
+  end
+  file clib => active_gems
+
   file mlib => clib
   file clib => init do |t|
     _pp "GEN", "*.rb", "#{clib.relative_path}"

